Mathlib/Data/Set/Basic.lean

```diff
@@ -193,6 +193,11 @@ theorem mem_of_mem_of_subset {x : α} {s t : Set α} (hx : x ∈ s) (h : s ⊆ t
 theorem forall_in_swap {p : α → β → Prop} : (∀ a ∈ s, ∀ (b), p a b) ↔ ∀ (b), ∀ a ∈ s, p a b := by
   tauto
 
+theorem setOf_injective : Function.Injective (@setOf α) := injective_id
+
+@[simp]
+theorem setOf_inj {p q : α → Prop} : { x | p x } = { x | q x } ↔ p = q := Iff.rfl
```

Comment:
This would cause `simp` do to the following:
```
example : {x : ℝ | x ^ 2 = 3} = {x : ℝ | x ^ 4 = 9 } := by
  simp
  -- ⊢ (fun x ↦ x ^ 2 = 3) = fun x ↦ x ^ 4 = 9
```
Is this desirable?

Reply:
I have to agree, this seems like quite a ludicrous simp lemma. It can definitely be useful as a simp lemma, but it shouldn't be tagged as one.